helpers/auth_helper.rb

```diff
@@ -0,0 +1,17 @@
+module AuthHelper
+  def auth_link
+    link_to title, url
+  end
+
+  def url
+    authorized ? '/profile' : '/auth/login'
+  end
+
+  def title
+    authorized ? 'Profile' : 'Login'
+  end
+
+  def authorized
+    @authorized ||= AuthenticationService::CurrentUser.call(env)
```

Comment:
Looks like you assign `current_user` just one time per controller_action there, so memoize inside service is unnecessary.
Also, I prefer to move this method inside application controller under `private` with access using `helper_method`, and rename method like `current_user`, because in Ruby web that's more clear naming for such a purpose. 
Helper plays role more as decorator, but not as presenter. And move that to presenter is not a solution, because you will need differentiate who works with system now logged user or guest in many places.
Investigate about a `helper_method` approach analogue for Sinatra, please.
What do you think about that? @kushniryb

Reply:
Makes perfect sense :+1: